src/Chapter2.hs

```diff
@@ -842,7 +847,8 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int] 
+rotate movementsToLeft numbers = take (length numbers) (drop movementsToLeft (cycle numbers))
```

Comment:
Unfortunately, `cycle` fails in runtime on empty lists :recycle: So you need to handle this case separately for this function to work properly.